includes/settings/js/src/components/fields/MediaFields.jsx

```diff
@@ -0,0 +1,49 @@
+/**
+ * Additional form fields for the Media field type.
+ */
+import React, { useState } from "react";
+import { __ } from "@wordpress/i18n";
+
+const MediaFields = ({ register, data, editing, fieldId, watch }) => {
+	const [showRepeatableMedia, setShowRepeatableMedia] = useState(
+		data?.isRepeatable === true
+	);
+
+	const [isFeatured, setIsFeatured] = useState(data?.isFeatured === true);
+
+	const isFeaturedWatcher = watch("isFeatured");
+
+	return (
+		<>
+			{data && (
+				<div className={"field"}>
+					<legend>
+						{__("Repeatable Field", "atlas-content-modeler")}
+					</legend>
+					<input
+						name="isRepeatableMedia"
+						type="checkbox"
+						id={`is-repeatable-${fieldId}`}
+						ref={register}
+						value={showRepeatableMedia}
+						onChange={() =>
+							setShowRepeatableMedia(!showRepeatableMedia)
+						}
+						disabled={isFeaturedWatcher}
```

Comment:
We should probably disable this when it's being edited to be consistent with the other “make repeatable” fields. (You can't change a field's repeatable status once it has been created.)

```suggestion
						disabled={editing || isFeaturedWatcher}
```